funcx_sdk/funcx/serialize/concretes.py

```diff
@@ -124,16 +129,93 @@ class DillCode(SerializeBase):
     def __init__(self):
         super().__init__()
 
-    def serialize(self, data):
+    def serialize(self, data) -> str:
         x = codecs.encode(dill.dumps(data), "base64").decode()
         return self.identifier + x
 
-    def deserialize(self, payload):
+    def deserialize(self, payload: str):
         chomped = self.chomp(payload)
         function = dill.loads(codecs.decode(chomped.encode(), "base64"))
         return function
 
 
+class CombinedCode(SerializeBase):
+    """This method uses multiple methods to serialize a function
+
+    # These should be a superset of the methods
+    Code from interpreter/main        : Yes
+    Code from notebooks               : Yes
+    Works with mismatching py versions: Yes
+    Decorated fns                     : Yes
+    """
+
+    identifier = "10\n"
+
+    def get_multiple_payloads(self, payload: str) -> tuple:
+        parts = self.chomp(payload).split(COMBINED_SEPARATOR)
+        assert len(parts) == 2 * len(DO_SERIALIZE_METHODS)
+        payloads: list[tuple[str, str]] = []
+        for index in range(len(parts) // 2):
+            payloads.append((parts[index], parts[index + 1]))
```

Comment:
Doesn't the range here need to be `range(0, len(parts) // 2, 2)`?  E.g. if we don't provide a step, then `len(parts) // 2 == 2` and index ranges over `[0,1]`.  That means if our parts look like `[04\n', 'gACQ...', '01\n', 'sAAbk2...']`, then we append `parts[0], parts[1]` in the first loop and `parts[1], parts[2]` in the second loop iteration.  That makes the result `((04\n', 'gACQ...'), ( 'gACQ...', '01\n')`.

You might should look at [itertools.pairwise](https://docs.python.org/3/library/itertools.html#itertools.pairwise).